spec/abstract/Structures_A.thy

```diff
@@ -383,24 +383,33 @@ datatype thread_state
   = Running
   | Inactive
   | Restart
-  | BlockedOnReceive obj_ref "obj_ref option" receiver_payload
+  | is_blocked_on_receive: BlockedOnReceive obj_ref (reply_object: "obj_ref option") receiver_payload
   | BlockedOnSend obj_ref sender_payload
   | BlockedOnReply obj_ref
   | BlockedOnNotification obj_ref
   | IdleThreadState
+  where
+  "reply_object (BlockedOnReply reply) = Some reply"
+  | "reply_object Running = None"
+  | "reply_object Inactive = None"
+  | "reply_object Restart = None"
+  | "reply_object (BlockedOnSend _ _) = None"
+  | "reply_object (BlockedOnNotification _) = None"
+  | "reply_object IdleThreadState = None"
```

Comment:
Would it be possible to use a default rule here? (I'm not sure any more which preprocessor is used for these)
```isabelle
"reply_object (BlockedOnReply reply) = Some reply"
   | "reply_object _ = None"
```

Reply:
I’d think yes, I’ll have a go.

Reply:
We can’t use the default rule here, actually.

Reply:
Ok, good to know.